app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/EnterPasscodeFragment.kt

```diff
@@ -54,9 +56,16 @@ class EnterPasscodeFragment : BaseViewBindingFragment<FragmentPasscodeBinding>()
 
     override fun onResume() {
         super.onResume()
-        binding.input.setRawInputType(InputType.TYPE_CLASS_NUMBER)
-        binding.input.delayShowKeyboardForView()
-        authenticateWithBiometrics()
+        if (alreadyStarted) {
+            findNavController().popBackStack(R.id.enterPasscodeFragment, true)
+            binding.input.hideSoftKeyboard()
+            alreadyStarted = false
+        } else {
+            alreadyStarted = true
+            binding.input.setRawInputType(InputType.TYPE_CLASS_NUMBER)
```

Comment:
why do we need to set input type in onResume? Can it be done in onViewCreated or even better in xml?

Reply:
I think we can remove that. It should be set in xml. But Liliya did not see the right keyboard. So I put it here too. But in the end it was her phone.